File: src/problems_all/217_contains_duplicate/contains_duplicate_v2.py
"""
Runtime:            3 ms
Beats:              98.38%
Memory:             31.54 MB
Beats:              23.44%
Submission:         https://leetcode.com/problems/contains-duplicate/submissions/1484689034/
Time complexity:    O(n)
Space complexity:   O(n)
Topics:             #array, #hash-table, #sorting
Solved By:          #hash-set
"""

from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)

        return False


def main():
    nums = [1, 2, 3, 4]
    result = Solution().containsDuplicate(nums)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/217_contains_duplicate/contains_duplicate_v2.go
/*
Runtime:            12 ms
Beats:              82.19%
Memory:             10.66 MB
Beats:              61.25%
Submission:         https://leetcode.com/problems/contains-duplicate/submissions/1484718403/
Time complexity:    O(n)
Space complexity:   O(n)
Topics:             #array, #hash-table, #sorting
Solved By:          #hash-table
*/

package main

import (
	"fmt"
)

func containsDuplicate(nums []int) bool {
	seen := make(map[int]bool)
	for _, num := range nums {
		if seen[num] {
			return true
		}
		seen[num] = true
	}

	return false

}

func main() {
	nums := []int{1, 2, 3, 4}
	result := containsDuplicate(nums)
	fmt.Println(result)
}
